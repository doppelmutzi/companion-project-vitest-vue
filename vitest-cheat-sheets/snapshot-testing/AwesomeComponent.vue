<template>
  <h1 id="awesome-component">{{ greeting }}</h1>
</template>

<script setup lang="ts">
import { computed } from "vue";

const props = defineProps<{ name: string }>();
const greeting = computed(() => "Hello, " + props.name);
</script>
<style scoped>
#awesome-component {
  color: red;
}
</style>
