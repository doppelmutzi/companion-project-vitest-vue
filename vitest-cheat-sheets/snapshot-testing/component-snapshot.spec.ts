import { mount } from "@vue/test-utils";
import { test, expect } from "vitest";
import AwesomeComponent from "./AwesomeComponent.vue";

test("renders component correctly", () => {
  const wrapper = mount(AwesomeComponent, {
    props: {
      name: "reader",
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});
