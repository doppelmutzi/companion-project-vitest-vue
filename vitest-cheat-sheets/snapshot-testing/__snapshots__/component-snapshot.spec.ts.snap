// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders component correctly 1`] = `"<h1 data-v-ff5b7346="" id="awesome-component">Hello, reader</h1>"`;
